platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/bls/protocol/ProtocolOutput.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.bls.protocol;
+
+/** An interface representing the output object of a {@link BlsProtocol} */
+public interface ProtocolOutput {
+    /**
+     * Compares this output object to another object. This comparison is more tolerant than a strict
+     * equals function, and should only compare object members that are expected to be the same for
+     * all parties completing a protocol. It shouldn't compare members that are secret to an
+     * individual node
+     *
+     * @param o the object to compare this one to
+     * @return true if the outputs are considered equal under the described relaxed conditions,
+     *     otherwise false
+     */
+    boolean compareToOtherOutput(Object o);
```

Comment:
If this is not an `.equals()` equivalent or overriding call, can the method parameter be a tighter type than `Object`?  Perhaps `ProtocolOutput`?

Reply:
@lpetrovic05   the record you commented on implements this interface.    Should records implement interfaces?

Reply:
records can implement interfaces but I do not get this one.

Reply:
No longer relevant https://github.com/hashgraph/hedera-services/pull/5559#discussion_r1137730303